src/DAMA/DoctrineTestBundle/Doctrine/DBAL/StaticConnection.php

```diff
@@ -41,7 +49,11 @@ public function beginTransaction(): bool
             throw new \BadMethodCallException(sprintf('Bad call to "%s". A savepoint is already in use for a nested transaction.', __METHOD__));
         }
 
-        $this->exec($this->platform->createSavePoint(self::SAVEPOINT_NAME));
+        try {
+            $this->exec($sql = $this->platform->createSavePoint(self::SAVEPOINT_NAME));
+        } catch (Driver\Exception $e) {
+            throw $this->driver->getExceptionConverter()->convert($e, new Query($sql, [], []));
+        }
```

Comment:
shall we add a private helper method `execQueryWithExceptionConversion` for this?